contributing/development/compiling/compiling_for_linuxbsd.rst

```diff
@@ -363,6 +363,71 @@ You don't even need to copy them, you can just reference the resulting
 files in the ``bin/`` directory of your Godot source folder, so the next
 time you build, you automatically have the custom templates referenced.
 
+Compiling for RISC-V
+--------------------
+
+To compile Godot for RISC-V, we need to setup the following items:
+
+- `riscv-gnu-toolchain <https://github.com/riscv-collab/riscv-gnu-toolchain/releases>`__.
+  While we are not going to use this directly, it provides us with a sysroot, as well
+  as header and libraries files that we will need. There are many versions to choose
+  from, however, the older the toolchain, the more compatible our final binaries will be.
+  If in doubt, `use this version <https://github.com/riscv-collab/riscv-gnu-toolchain/releases/tag/2021.12.22>`__,
+  and download ``riscv64-glibc-ubuntu-18.04-nightly-2021.12.22-nightly.tar.gz``. Extract
+  it somewhere and remember its path.
+- Clang. RISC-V GCC has
+  `bugs with its atomic operations <https://github.com/riscv-collab/riscv-gcc/issues/15>`__
+  which prevent it from compiling Godot correctly. Any version of Clang from 16.0.0 upwards
+  will suffice. Download it from the package manager of your distro, and make sure that
+  it *can* compile to RISC-V. You can verify by executing this command ``clang -print-targets``,
+  make sure you see ``riscv64`` on the list of targets.
+- `mold <https://github.com/rui314/mold/releases>`__. While ``lld`` can also be used, ``mold``
+  does a better job linking the final binary. Download it, extract it, and make sure
+  to add its ``bin`` folder to your PATH.
+
+With all the above setup, we are now ready to build Godot.
+
+Go to the root of the source code, and execute the following build command:
+
+::
+
+    scons arch="rv64" use_llvm="yes" linker="mold" lto="none" \
+        target="editor" \
+        module_theora_enabled="no" \
+        module_denoise_enabled="no" \
+        module_raycast_enabled="no" \
+        module_xatlas_unwrap_enabled="no" \
```

Comment:
The list of disabled modules shouldn't be necessary, because those modules should disable themselves based on the `config.py` file's `def can_build(env, platform):` function.

Reply:
It looks like `theora` already disables itself when compiling for RISC-V, I'll make a PR for `xatlas`, `raycast`, and `denoise` (not where that one is).

Reply:
I compiled 4.3-stable both the editor and export templates, they seem to compile just fine without having to disable those modules manually. I will update accordingly.